lib/active_model/serializer/array_serializer.rb

```diff
@@ -21,11 +22,11 @@ def initialize(objects, options = {})
       end
 
       def json_key
-        @objects.first.json_key if @objects.first
-      end
-
-      def root=(root)
-        @objects.first.root = root if @objects.first
+        if root == true && @objects.first
+          @objects.first.class.root_name
```

Comment:
We could define a default ```data``` root by default is there isn't one defined. But I'm not convinced yet about the root implementation on ```ArraySerializer``` :stuck_out_tongue_closed_eyes: